lib/dereference.js

```diff
@@ -125,6 +126,10 @@ function dereference$Ref ($ref, path, pathFromRoot, parents, $refs, options) {
     dereferencedValue.$ref = pathFromRoot;
   }
 
+  if (options.dereference.saveOriginalRefs) {
+    setMeta(dereferencedValue, '$ref', $ref.$ref);

```

Comment:
Let's add `path` and `pathFromRoot` as well.  I can see those being handy for folks (myself included 😊).   I suggest modifying the `setMetadata()` function to accept multiple properties at once.  That saves us from having to call it multiple times.

```suggestion
    setMetadata(dereferencedValue, {
        $ref: $ref.$ref,
        path: path,
        pathFromRoot: pathFromRoot,
    });
```